### PR TITLE
feat(linter): warn unmatched rule names

### LINKED ISSUE #6763: linter: warn unmatched rule names
Given 

```
  "rules": {
    "for-direction": "error",
    "no-async-promise-executor": "error",
    "no-caller": "error",
    "no-case-declarations": "error",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
...
```

I don't know which rule is applied or not, we should warn the ones that are incorrect.

### PR DESCRIPTION
**Pull request by tapanprakasht**
Fixes https://github.com/oxc-project/oxc/issues/6763

### CODE REVIEW

File: crates/oxc_linter/src/config/rules.rs [hunk 1]
```diff
@@ -121,6 +126,13 @@ impl OxlintRules {
         for rule in rules_to_replace {
             rules_for_override.replace(rule);
         }
+
+        if !rules_not_matched.is_empty() {
+            println!("The following rules do not match the currently supported rules:");
```
Comment (DonIsaac) [inline crates/oxc_linter/src/config/rules.rs:131]: Consider using `OxcDiagnostic::warn` and the reporter instead of directly printing to stdout.

**Review by DonIsaac — DISMISSED** [review]

**Review by Boshen — CHANGES_REQUESTED** [review]
Consider using OxcDiagnostic::warn and the reporter instead of directly printing to stdout.

File: crates/oxc_linter/src/config/rules.rs (hunk 1 above)
Comment (tapanprakasht) [reply crates/oxc_linter/src/config/rules.rs:131]: What is the reporter here? any example from code base?

Comment (Boshen) [reply crates/oxc_linter/src/config/rules.rs:131]: Oh ...  surfacing the warning is actually harder than I thought.

**Review by Boshen — APPROVED** [review]
Let's merge and iterate.

**Pull request merged**